### PR TITLE
Use the appropriate type for create server methods

### PR DESCRIPTION
**Pull request by jrichter1**
so they match the RSP server's response

### CODE REVIEW

**Comment by adietish** [comment]
@jrichter1 @jeffmaury as it turns out these kind of required changes are only visible once we have the client actually talk to a server in integration/manual tests. This boils down to the fact that only the transferred parameters are generated to json/ts. If the requests and their responses were we had an error right in the compiler.

ex. RSPClient#createServerAsync now gets a CreateServerResponse back while it was a Status before this change:
```
-    createServerAsync(pathOrBean: string | Protocol.ServerBean, id?: string, attributes?: { [index: string]: any }, timeout: number = 2000): Promise<Protocol.Status> {
+    createServerAsync(pathOrBean: string | Protocol.ServerBean, id?: string, attributes?: { [index: string]: any }, timeout: number = 2000): Promise<Protocol.CreateServerResponse> {
```

```ServerModel#createServerFromPathAsync``` sends a ```CreateServerRequest``` to achieve this:
```
return Common.sendSimpleRequest(this.connection, Messages.Server.CreateServerRequest.type, serverAttributes,
            timeout, ErrorMessages.CREATESERVER_TIMEOUT);
```
The ```CreateServerRequest``` has a notion of the return type, the ```Protocol.Status``` parameter:
```
        export namespace CreateServerRequest {
            export const type = new RequestType<Protocol.ServerAttributes, Protocol.Status, void, void>('server/createServer');
        }
```

The fact to the matter is that since it's not included in the classes that are generated as schema in the server results in the fact that "installing" the new schema wont error at compile time in RSPClient.
We could easily fix this by having the server also generating the requests.
Thoughts?

**Comment by jrichter1** [comment]
That sounds a bit too implementation specific to me. Generating this would essentially lock any ts-based client into using vscode-jsonrpc library for the underlying connection.
The potential solution would be to generate the API and then the entire client.

File: test/serverModel-test.ts [hunk 1]
```diff
@@ -33,6 +33,10 @@ describe('Sever Model Utility', () => {
         severity: 0,
         trace: 'trace'
     };
+    const createStatus: Protocol.CreateServerResponse = {
+        status: status,
+        invalidKeys: []
```
Comment (jeffmaury) [inline test/serverModel-test.ts]: Any reason why it's not typed ?

Comment (jrichter1) [reply test/serverModel-test.ts]: What exactly is not typed here?

**Comment by adietish** [comment]
@jrichter1 I see that request and responses are currently bound to vscode-jsonrpc but then no one is forced to use those. Any potential client is free to use those too or not. Furthermore imho let's be honest: how likely is it that anyone else would create a different client if there's one ready at hand? Chances are very small. Typesafety is to me the far bigger gain here if being bound was at stake which it isn't I believe.
And then, the additional option to generate the client is even more compelling on the long run. I believe that we should go this route on the long run anyhow.

File: test/serverModel-test.ts (hunk 1 above)
Comment (jeffmaury) [reply test/serverModel-test.ts]: The invalidKeys array

Comment (jrichter1) [reply test/serverModel-test.ts]: invalidKeys type is inferred from here:
https://github.com/redhat-developer/rsp-client/blob/master/src/protocol/protocol.ts#L29

This is just an assignment of empty array into a string array.

Comment (adietish) [reply test/serverModel-test.ts]: @jrichter1 so why not type it here, too?

Comment (jrichter1) [reply test/serverModel-test.ts]: because this is an assignment into an already typed variable

**Comment by adietish** [comment]
@jrichter1 thx for the patch. Nevertheless, I believe that we should think hard whether we should generate request and responses type so that we have compile-time errors for server API changes.

**Pull request merged**